### PR TITLE
Added searching algorithms in Java...

### LINKED ISSUE #33: [JAVA] : Add Searching Algorithms
Add Searching Algorithms like binary search, linear search, sentinel search etc.

If interested, please add the algorithms into `java/searching` package

### PR DESCRIPTION
**Pull request by Dnyanesh-Bachhav**
Resolved #33
Added searching algorithms in Java...

### CODE REVIEW

File: java/searching/linear_search.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        {
+            return arr1[i];
+        }
+    }
+    return -1;
+    }
+
+    public static void main(String args[])
+    {
+        int arr1[] = {10,8,12,98,67};
```
Comment (ShubhKotnala) [inline java/searching/linear_search.java:16]: C style declaration of array

File: java/searching/linear_search.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+public class linear_search{
```
Comment (ShubhKotnala) [inline java/searching/linear_search.java:1]: Class should follow Java class naming convention

File: java/searching/binary_search.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+public class binary_search{
+
+    static int binarySearch(int[] arr1, int key){
+    int start = 0;
+    int end = arr1.length - 1;
+
+    while (start <= end) {
+        int middle = (int)Math.floor((start + end) / 2);
+        System.out.println("middle: "+middle);
```
Comment (ShubhKotnala) [inline java/searching/binary_search.java:9]: Is this required?

File: java/searching/binary_search.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            end = middle - 1;
+        }
+    }
+	// key wasn't found
+    return -1;
+    }
+
+    public static void main(String args[])
+    {
+        int arr1[] = {10,20,30,40,50};
```
Comment (ShubhKotnala) [inline java/searching/binary_search.java:28]: C style declaration of array

File: java/searching/binary_search.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
+public class binary_search{
```
Comment (ShubhKotnala) [inline java/searching/binary_search.java:1]: Follow Java class naming conventions

**Review by ShubhKotnala — CHANGES_REQUESTED** [review]
File and class names should follow Java class naming conventions. 
Also, please format the code so it looks much more organised.

**Comment by ShubhKotnala** [comment]
Also, do update the file with a comment block explaining what problem your code solves along with some test cases

File: java/searching/binary_search.java (hunk 3 above)
Comment (Dnyanesh-Bachhav) [reply java/searching/binary_search.java]: Yes, actually Math.floor() returns a double value to convert it into int.. requires conversion...

File: java/searching/linear_search.java (hunk 2 above)
Comment (Dnyanesh-Bachhav) [reply java/searching/linear_search.java]: Now in the recent commit followed all naming conventions...

File: java/searching/binary_search.java (hunk 4 above)
Comment (Dnyanesh-Bachhav) [reply java/searching/binary_search.java]: It works fine... Do I need to change it??

**Review by ShubhKotnala — APPROVED** [review]

**Pull request merged**